mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockResponse.java

```diff
@@ -130,15 +128,20 @@ public MockResponse removeHeader(String name) {
     return this;
   }
 
-  /** Returns the raw HTTP payload, or null if this response is streamed. */
-  public Buffer getBody() {
-    return body != null ? body.clone() : null; // Defensive copy.
+  /** Returns the raw HTTP payload. */
+  public Source getBody() {
+    return body;
   }
 
-  Source getBodySource() {
-    return bodyStream != null ? Okio.source(bodyStream) : getBody();
+  /**
+   * The number of bytes in {@linkplain #getBody() the body}. Note: this is not synonymous with
+   * {@code Content-Length} (e.g., in the case of chunked body).
+   */
+  public long getBodySize() {
```

Comment:
Looks like we only ever compare this to 0. Perhaps instead we could just have a null response from body source? The fact that this doesn't line up with `Content-Length` is ugly.